Apps/phhackerone/hackerone_connector.py

```diff
@@ -0,0 +1,689 @@
+            self.__print(err)
+            return None
+
+    def _add_report_artifact(self, report):
+        self.__print('_add_report_artifact()')
+        artifact = {}
+        artifact['container_id'] = self.get_container_id()
+        artifact['label'] = 'Report'
+        artifact['name'] = 'HackerOne Report - {0}'.format(report['id'])
+        artifact['source_data_identifier'] = report['id']
+        artifact['severity'] = 'medium'
+        artifact['cef'] = report
+        self.save_artifact(artifact)
+
+    def _add_report_artifacts(self, reports):
+        self.__print('_add_report_artifacts()')
+        artifacts = []
+        for report in reports:
+            cef = report
+            artifact = {}
+            artifact['container_id'] = self.get_container_id()
+            artifact['label'] = 'Report'
+            artifact['name'] = 'HackerOne Report - {0}'.format(cef['id'])
+            artifact['source_data_identifier'] = '{0}-{1}'.format(cef['id'], self.get_container_id())
+            artifact['severity'] = 'medium'
+            artifact['cef'] = cef
+            artifacts.append(artifact)
+        self.save_artifacts(artifacts)
+
+    def _update_tracking_id(self, param, action_result):
+        self.__print('_update_tracking_id()')
+        report_id = self._handle_unicode_for_input_str(param.get('report_id'))
+        tracking_id = self._handle_unicode_for_input_str(param.get('tracking_id'))
+        try:
+            data = {
+                "data": {
+                    "type": "issue-tracker-reference-id",
+                    "attributes": {
+                        "reference": tracking_id
+                    }
+                }
+            }
+            url = "https://api.hackerone.com/v1/reports/" + report_id + "/issue_tracker_reference_id"
+            if self._post_rest_data(url, data):
+                self.__print('Successfully updated tracking id')
+                return action_result.set_status(phantom.APP_SUCCESS, 'Successfully updated tracking id')
+            else:
+                self.__print('Failed to update tracking id.')
+                return action_result.set_status(phantom.APP_ERROR, 'Failed to update tracking id')
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            self.__print('Exception occurred while updating tracking id')
+            action_result.add_exception_details(err)
+            return action_result.set_status(phantom.APP_ERROR, 'Exception occurred while updating tracking id')
+
+    def _unassign_report(self, param, action_result):
+        self.__print('_unassign_report()')
+        report_id = self._handle_unicode_for_input_str(param.get('report_id'))
+        try:
+            data = {
+                "data": {
+                    "type": "nobody"
+                }
+            }
+            url = "https://api.hackerone.com/v1/reports/" + report_id + "/assignee"
```

Comment:
can we use `.format` instead of string concatenation?

Reply:
Fixed